hyperspy/model.py

```
@@ -891,6 +899,312 @@ def _model_function(self, param):
+                "one component. "
+                "This is not recommended for linear optimizers, since the optimizer is unable "
+                "to determine if there are any parts of the component that are constant. "
+                "Example: comp = a*x+b, with b.free = False and non-zero. "
+                "Either continue at own risk, or define the component as a `Expression` instead."
+                )
+
+        self._set_linear_parameters_to_one()
+        self._set_p0()
+
+        channels_signal_shape = np.count_nonzero(self.channel_switches)
+
+        self._component_data = np.zeros((n_free_para, channels_signal_shape))
+        self._component_data_fixed = np.zeros(channels_signal_shape)
+
+        self._set_twinned_lists()
+        for component in self:
+            # only compute if component and topmost parent are active
+            if component.active and check_top_parent_twins_are_active(component):
+                self._compute_component(component)
+
+        if self._linear_ndfit:
+            nav_shape = self.axes_manager._navigation_shape_in_array
+            if self.channel_switches.all():
+                # If channel_switches is all True, then is much more performant
+                # and less memory-intensive to just reshape the signal than use 
+                # fancy indexing (which would create a copy of the signal),
+                # especially with dask
+                target_signal = self.signal.data.reshape(nav_shape + (-1,))
+            else:
+                target_signal = self.signal.data.T[np.where(self.channel_switches.T)].T
+        else:
+            target_signal = self.signal()[np.where(self.channel_switches)]
+            nav_shape = ()
+
+        if is_binned(self.signal):
+            target_signal = target_signal / np.prod(
+                tuple((ax.scale for ax in self.signal.axes_manager.signal_axes))
+            )
```

It implies that linear fitting doesn't support *binned* and *non-uniform* axis? If so, for now it should raise an error. Support for binned non-uniform axis can be added later.

---

Yes, this is correct, good catch.